core/arch/arm/plat-stm32mp1/plat_huk.c

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Linaro Limited
+ * Copyright (c) 2022, Foundries.io Limited
+ */
+
+#include <assert.h>
+#include <config.h>
+#include <crypto/crypto.h>
+#include <drivers/stm32_bsec.h>
+#include <kernel/tee_common_otp.h>
+#include <platform_config.h>
+#include <string.h>
+
+#if !defined(CFG_STM32MP1_HUK_BSEC_BASE)
+#error "CFG_STM32MP1_HUK_BSEC_BASE undefined"
+#endif
+
+static TEE_Result aes_gcm_encrypt_uid(uint8_t *key, size_t key_len,
+				      uint8_t *out, size_t *out_len)
+{
+	uint8_t uid[HW_UNIQUE_KEY_LENGTH] = { 0 };
+	uint8_t nonce[12] = {
+		0xd2, 0x45, 0x0e, 0x07, 0xea, 0x5d,
+		0xe0, 0x42, 0x6c, 0x0f, 0xa1, 0x33,
+	};
+	uint8_t tag[16] = { 0 };
+	TEE_Result res = TEE_ERROR_GENERIC;
+	size_t tag_len = sizeof(tag);
+	size_t uid_len = sizeof(uid);
+	uint32_t word = 0;
+	void *ctx = NULL;
+	size_t n = 0;
+
```

Comment:
for consistency:
`static_assert(HW_UNIQUE_KEY_LENGTH >= (UID2_OTP - UID0_OTP + 1) * sizeof(uint32_t));`